cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSLAHttpHandler.java

```diff
@@ -0,0 +1,99 @@
+import co.cask.cdap.internal.app.store.WorkflowDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Workflow Statistics Handler
+ */
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}")
+public class WorkflowStatsSLAHttpHandler extends AbstractHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WorkflowStatsSLAHttpHandler.class);
+  private final Store store;
+
+  @Inject
+  WorkflowStatsSLAHttpHandler(Store store) {
+    this.store = store;
+  }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/statistics")
+  public void workflowStats(HttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId,
+                            @PathParam("app-id") String appId,
+                            @PathParam("workflow-id") String workflowId,
+                            @QueryParam("start") String start,
+                            @QueryParam("end") String end,
+                            @QueryParam("percentile") List<Double> percentiles) throws Exception {
+
+    long startTime = TimeMathParser.parseTimeInSeconds(start);
+    long endTime = TimeMathParser.parseTimeInSeconds(end);
+
+    if (startTime < 0) {
+      throw new BadRequestException("Wrong start time.");
+    } else if (endTime < 0) {
+      throw new BadRequestException("Wrong end time.");
+    } else if (endTime < startTime) {
+      throw new BadRequestException("Start time cannot be larger than end time.");
+    }
+
+    for (double i : percentiles) {
+      if (i < 0.0 || i > 100.0) {
+        throw new BadRequestException("Percentile values have to be greater than or equal to 0 and" +
+                                        " less than or equal to 100. Invalid input was " + Double.toString(i));
+      }
+    }
+
+    WorkflowDataset.BasicStatistics basicStatistics;
+
+    Id.Workflow workflow = Id.Workflow.from(Id.Namespace.from(namespaceId), appId, workflowId);
+    try {
+      basicStatistics = store.getWorkflowStatistics(workflow, startTime, endTime, percentiles);
+    } catch (IllegalArgumentException exception) {
+      throw new BadRequestException(exception);
+    }
+    if (basicStatistics == null) {
+      responder.sendString(HttpResponseStatus.OK, "There are no statistics associated with this workflow : "
+        + workflowId + " in the specified time range.");
+      return;
+    }
+    responder.sendJson(HttpResponseStatus.OK, basicStatistics);
```

Comment:
If you are returning ``BasicStatistics`` as an object from a REST endpoint, then the ``BasicStatistics`` class should move to ``cdap-proto``. This is so clients can parse the response using that object. It shouldn't be an inner class. Also, lets name it ``WorkflowStatistics`` or something like that.